store/postgres/src/subgraph_migrator.rs

```diff
@@ -0,0 +1,56 @@
+use graph::components::store::DeploymentSchemaVersion;
+use graph::prelude::StoreError;
+
+use crate::deployment_store::DeploymentStore;
+use crate::subgraph_store::SubgraphStore;
+use crate::{dynds::DataSourcesTable, primary::Site};
+use std::sync::Arc;
+
+pub struct SubgraphMigrator<'a> {
+    site: Arc<Site>,
+    subgraph_store: &'a SubgraphStore,
+    store: Arc<DeploymentStore>,
+    current_version: DeploymentSchemaVersion,
+}
+
+impl<'a> SubgraphMigrator<'a> {
+    pub fn new(
+        site: Arc<Site>,
+        subgraph_store: &SubgraphStore,
+        store: Arc<DeploymentStore>,
+    ) -> SubgraphMigrator {
+        return SubgraphMigrator {
+            current_version: site.schema_version,
+            subgraph_store: &subgraph_store,
+            store,
+            site,
+        };
+    }
+
+    pub fn run(mut self) -> Result<(), StoreError> {
+        if self.current_version == DeploymentSchemaVersion::V1 {
+            self.migrate_to_v2()?;
+        }
+        Ok(())
+    }
+
+    fn migrate_to_v2(&mut self) -> Result<(), StoreError> {
+        let migration = format!(
+            "
+        alter table {table}
+        add done bool not null default false;
```

Comment:
On the name, I find "processed" a bit ambiguous between partially processed or fully processed. Also "an onchain data source is never processed" doesn't sound right. Open to other ideas: `finished`, `completed`, `finalized`...

I agree that we can make this this `done_at`, but store the block number on which it became done rather than the timestamp.

Reply:
I am totally fine with calling this `finished` or the like. Even better might be a comment on `DataSource.done_at` that explains how that flag is used, and what it means. If we have a clear explanation there, I am also fine with `done`.

The reason I wanted a timestamp there rather than the block number is so that indexers can get a sense for how long ago (in wall time) that data source was processed. The motivation for this is that in some situations it would be quite useful with a subgraph's `synced` status to see how long ago that was reached. If storing a block number there implies that that was the block number that was the chain head at the time the data source was finished.

Reply:
I don't feel strongly on the name. @poonai do you have opinions? Agreed on improving the comments.

While I don't expect the block number to ever be deterministic it might be more reproducible and comparable across indexers than timestamps. Anyways this value might never be used at all, it isn't yet clear how the indexing status of offchain data will be reported, so lets not overthink.

Reply:
I really don't have any strong opinion on this.